lib/internal/modules/cjs/loader.js

```diff
@@ -829,9 +829,19 @@ Module.runMain = function() {
 };
 
 Module.createRequireFromPath = (filename) => {
-  const m = new Module(filename);
-  m.filename = filename;
-  m.paths = Module._nodeModulePaths(path.dirname(filename));
+  // Allow a directory to be passed as the filename
+  const normalized = path.win32.normalize(filename);
+  const trailingSlash = normalized.charCodeAt(normalized.length - 1) ===
+    CHAR_BACKWARD_SLASH;
```

Comment:
Should we maybe just call `stat` to verify if it's a file or directory? That way it's even safer and works without a trailing slash.

Reply:
I think the issue isn't being different from module it is being different from how Path works... I'm not 100% we should be adding the stat here